File: src/main/java/com/listaai/infrastructure/adapter/output/persistence/mapper/ListPersistenceMapper.java
package com.listaai.infrastructure.adapter.output.persistence.mapper;

import com.listaai.domain.model.ShoppingList;
import com.listaai.infrastructure.adapter.output.persistence.entity.ListEntity;
import org.mapstruct.Mapper;

@Mapper(componentModel = "spring")
public interface ListPersistenceMapper {
    ShoppingList toDomain(ListEntity entity);
    ListEntity toEntity(ShoppingList domain);
}

File: src/main/kotlin/com/listaai/infrastructure/adapter/output/persistence/mapper/ListPersistenceMapper.kt
package com.listaai.infrastructure.adapter.output.persistence.mapper

import com.listaai.domain.model.ShoppingList
import com.listaai.infrastructure.adapter.output.persistence.entity.ListEntity
import org.mapstruct.Mapper

@Mapper(componentModel = "spring")
interface ListPersistenceMapper {

    fun toDomain(list: ListEntity): ShoppingList
    fun toEntity(list: ShoppingList): ListEntity
}